pulp_container/app/tasks/sync_stages.py

```diff
@@ -349,6 +379,117 @@ def create_blob(self, blob_data, deferred_download=True):
 
         return blob_dc
 
+    async def create_signatures(self, man_dc, signature_source):
+        """
+        Create signature declarative artifacts.
+
+        Signatures are currently supported only for image manifests and not manifest lists.
+
+        For sigstore, signatures are downloaded from a specified url. The number of signatures is
+        unknown upfront, need to download them in order by incrementing the index until hit 404.
+
+        Args:
+            man_dc: Declarative content instance of a related Manifest
+            signature_source: Source where to get signatures from
+        """
+
+        def _extract_data_from_signature(signature_raw):
+            gpg = gnupg.GPG()
+            crypt_obj = gpg.decrypt(signature_raw)
+            if not crypt_obj.data:
+                log.info(
+                    _(
+                        "It is not possible to read the signed document, GPG error: {}".format(
+                            crypt_obj.stderr
+                        )
+                    )
+                )
+                return
+
+            try:
+                sig_json = json.loads(crypt_obj.data)
```

Comment:
we should add signature validation of the loaded json

Reply:
Please elaborate what kind of validation you expect?

Reply:
we should assert that everything what's in 'critical' is there and refuse when unrecognized or unsupported is encountered https://github.com/containers/image/blob/main/docs/containers-signature.5.md#json-processing-and-forward-compatibility

Reply:
let's capture this in a separate ticket and implement later not in mvp

Reply:
https://pulp.plan.io/issues/9597